File: 451-sort-characters-by-frequency/sort-characters-by-frequency.java
class Solution {
    public String frequencySort(String s) {

        HashMap<Character, Integer> map = new HashMap<>();

        for (char key : s.toCharArray()) {
            if (map.containsKey(key)) {
                int freq = map.get(key);
                map.put(key, freq + 1);
            } else {
                map.put(key, 1);
            }
        }

        List<Map.Entry<Character, Integer>> list = new ArrayList<>(map.entrySet());

        list.sort(Map.Entry.<Character, Integer>comparingByValue().reversed());

        StringBuilder ans = new StringBuilder();

        for (Map.Entry<Character, Integer> entry : list) {

            char ch = entry.getKey();
            int freq = entry.getValue();

            for (int i = 0; i < freq; i++) {
                ans.append(ch);
            }
        }

        return ans.toString();
    }
}

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.py
class Solution(object):
    def frequencySort(self, s):
        map = {}
        for key in s:
            if key in map:
                map[key] += 1
            else:
                map[key] = 1
        result = list(map.items())
        result.sort(key=lambda x: x[1], reverse=True)

        str = ''
        for index in range(len(result)):
            value = result[index][0]
            freq = result[index][1]
            for i in range(freq):
                str = str + value
        return str